### ISSUE TITLE
Wrong version

### ISSUE THREAD

**Issue by OskarStark** [dependabot/dependabot-core#161]
This is written down in my `composer.json`

`"sensio/framework-extra-bundle": "^3.0.2",`

this means, that everything below `3.1` is allowed. Why do I get a PR like this?
![screenshot 2017-12-08 11 50 45](https://user-images.githubusercontent.com/995707/33762762-0c0af930-dc0e-11e7-8f64-de9cc241bf4c.png)

**Comment by OskarStark** [comment]
It could create many version PR's ofc, like:

- `3.0.36` -> `3.0.37` (_just an example, don't know yet if this version exists_)
- `3.0.36` -> `4.0.0`
- `3.0.36` -> `5.0.21`

to see if the tests fail or not

**Comment by greysteil** [comment]
Here's the thinking on this one:
- Dependabot creates PRs to get you to the latest version that your `composer.json` can resolve to.
- It **doesn't** honour the specified requirements in your `composer.json` for a couple of reasons:
  - people specify constraints in very different ways (some tightly, some loosely), often without thinking through the consequences in great detail. In many cases, they'd get unexpected results from Dependabot if we *did* honour the constraints
  - we hope it's generally useful to get a reminder when a new major version is released, even if you don't update to it
  - most dependencies have a single supported major version, and aren't releasing patch versions for older releases (the exceptions are obviously the really big frameworks)

The above definitely isn't perfect
- it can be a pain when you're first using Dependabot, as you probably want to upgrade one major version at a time. We should definitely do something to make that easier, but don't support it yet.
- it can also be a pain if you really don't want to upgrade, for whatever reason. Dependabot has an `ignore this major version` feature that hopefully helps handle that one.

Basically, it's what we've got for now, but we can definitely do better in future. I'm going to close this, but will keep it in mind and you're welcome to open a feature request on https://github.com/dependabot/feedback.

**Comment by OskarStark** [comment]
Thank you for the detailed feedback 👍

**Comment by greysteil** [comment]
@OskarStark - there are actually a couple of improvements on this one since it was originally opened:
- You can now ask Dependabot not to update your `composer.json` in the update config settings in your account dashboard (click the down arrow next to "Bump now").
- For some languages (Java and Python) Dependabot will still give you updates to previous major version lines if you've asked to ignore a subsequent major version. I could roll that out to PHP relatively straightforwardly if it would help?

**Comment by greysteil** [comment]
👍 - if I get lots of requests to do that I'll switch over. I've also implemented the ability for Dependabot to respect ignore conditions (specified by commenting `@dependabot ignore this major version`) when generating updates in [this PR](https://github.com/dependabot/dependabot-core/pull/497), which should be good to deploy today.

Thanks for the feedback, as ever! 😄
